project/BuildPlugin.scala

```diff
@@ -38,23 +40,36 @@ object BuildKeys {
   // Use absolute paths so that references work even if `ThisBuild` changes
   final val AbsolutePath = file(".").getCanonicalFile.getAbsolutePath
 
-  final val ZincProject = RootProject(file(s"$AbsolutePath/zinc"))
+  private val isCiDisabled = sys.env.get("CI").isEmpty
+  def createScalaCenterProject(name: String, f: File): RootProject = {
+    if (isCiDisabled) RootProject(f)
+    else {
+      val headSha = new com.typesafe.sbt.git.DefaultReadableGit(f).withGit(_.headCommitSha)
+      headSha match {
+        case Some(commit) => RootProject(uri(s"git://github.com/scalacenter/${name}.git#$commit"))
+        case None => sys.error(s"The 'HEAD' sha of '${f}' could not be retrieved.")
+      }
+    }
+  }
+
+  final val ZincProject = createScalaCenterProject("zinc", file(s"$AbsolutePath/zinc"))
```

Comment:
Wow nice!

Reply:
First time I realized we could do this, but yeah it's actually a great mechanism to handle this :smile:.